File: .app/lib/models/vaccine_dose.dart
class VaccineDose {
  VaccineDose({
    required this.id,
    required this.name,
    required this.doseNumber,
    required this.scheduledDate,
    this.completedDate,
  });

  final String id;
  final String name;
  final String doseNumber;
  final DateTime scheduledDate;
  final DateTime? completedDate;

  factory VaccineDose.fromJson(Map<String, dynamic> json) => VaccineDose(
        id: json['id'] as String,
        name: json['vaccineName'] as String,
        doseNumber: json['doseNumber'] as String,
        scheduledDate: DateTime.parse(json['scheduledDate'] as String),
        completedDate: json['completedDate'] == null ? null : DateTime.parse(json['completedDate'] as String),
      );
}

File: .backend/Dopa.Api/Models/VaccineDose.cs
using System;

namespace Dopa.Api.Models;

public class VaccineDose : BaseEntity
{
    public Guid UserId { get; set; }
    public string VaccineName { get; set; } = string.Empty;
    public string DoseNumber { get; set; } = string.Empty;
    public DateTime ScheduledDate { get; set; }
    public DateTime? CompletedDate { get; set; }

    public User? User { get; set; }
}
